### PR TITLE
for-review

### PR DESCRIPTION
**Pull request by christina-mifsud**
Completed Level 100 as per instructions

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -1,12 +1,121 @@
...
 
 function makePageForEpisodes(episodeList) {
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  //rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+
+  // level 100
+  document.getElementById("root").innerHTML = "";
+  for (i = 0; i < episodeList.length; i++) {
+    // step 1 - create elements
```
Comment (nfarah22) [inline script.js:152]: i really like how you have used comments on each stage of your code. it makes it much easier to follow your code and its it going something i will try to implement on my project.

File: script.js [hunk 2]
```diff
@@ -1,12 +1,121 @@
...
 }
 
 function makePageForEpisodes(episodeList) {
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  //rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+
+  // level 100
+  document.getElementById("root").innerHTML = "";
+  for (i = 0; i < episodeList.length; i++) {
```
Comment (nfarah22) [inline script.js:151]: i find it easier to use array methods. like forEach is a good method to use here.

File: script.js [hunk 3]
```diff
@@ -1,12 +1,121 @@
...
+//makePageForEpisodes(allEpisodes); // OG
+
+// level 200 - search bar
+
+document
+  .getElementById("search-input")
+  .addEventListener("input", searchEpisode);
+
+function searchEpisode() {
+  const searchInput = document
```
Comment (nfarah22) [inline script.js:221]: its really interesting that you made another function for this task. i didnt even think of that .

File: script.js [hunk 4]
```diff
@@ -1,12 +1,121 @@
...
+    // assign unique ID to individual episode-container
+    episodeDivContainer.id = "episode-container" + i;
+
+    // step 2 - contents of elements
+    episodeTitleElement.innerHTML = `${episodeList[i].name}-S${String(
+      episodeList[i].season
+    ).padStart(2, "0")}E${String(episodeList[i].number).padStart(2, "0")}`;
+    episodeImageElement.src = `${episodeList[i].image.medium}`;
+    episodeBlurbElement.innerHTML = `${episodeList[i].summary}`;
+
```
Comment (nfarah22) [inline script.js:173]: i found this part of your code hard to follow. i had to google why you were using padStart and what it does.this was a was a great learning experience for me so great work.

File: index.html [hunk 5]
```diff
@@ -3,15 +3,37 @@
...
+      <select name="episodes-list" id="episode-select"></select>
     </div>
 
+    <div id="search-bar">
+      <input
+        id="search-input"
+        type="search"
+        placeholder="Search for an episode"
+      />
+      <label for="search"
```
Comment (msimmdev) [inline index.html]: Issue: The `for` attribute of label must match an id of an input element. There is no input with id 'search' on the page.

File: script.js [hunk 6]
```diff
@@ -1,12 +1,121 @@
...
+function fetchAllEpisodes() {
+  return fetch("https://api.tvmaze.com/shows/82/episodes")
+    .then((response) => response.json())
+    .then((data) => {
+      allEpisodes = data; // this was done to move allEpisodes to global scope
+      makePageForEpisodes(allEpisodes);
+    })
+    .catch(function (error) {
+      console.log(error);
```
Comment (msimmdev) [inline script.js]: Praise: Great to have a catch that logs your errors.
Suggestion: Using `console.error()` instead of `console.log()` will make the browser treat the error as an error which can help debugging.

File: script.js [hunk 7]
```diff
@@ -1,12 +1,121 @@
...
+    });
+}
+
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  fetchAllEpisodes();
```
Comment (msimmdev) [inline script.js]: NitPick: The `setup()` function is doing nothing other than calling the `fetchAllEpisodes()` function. There's no advantage to having two functions in this case.

File: script.js (hunk 2 above)
Comment (msimmdev) [reply script.js:151]: Nitpick: I disagree that `forEach` would be better here as there's no chaining with other array methods required, however I'd prefer to see a `for .. of ` loop, as this is more readable than a simple `for` loop or a `forEach`.

File: script.js [hunk 8]
```diff
@@ -1,12 +1,121 @@
...
+  // makePageForEpisodes(allEpisodes);
 }
 
 function makePageForEpisodes(episodeList) {
   const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  //rootElem.textContent = `Got ${episodeList.length} episode(s)`;
```
Comment (msimmdev) [inline script.js]: Nitpick: Always avoid leaving commented code in your solution.

File: script.js [hunk 9]
```diff
@@ -1,12 +1,121 @@
...
+    let episodeDivContainer = document.createElement("div");
+    let episodeTitleElement = document.createElement("h3");
+    let episodeImageElement = document.createElement("img");
+    let episodeBlurbElement = document.createElement("p");
+
+    // assign the divs classes
+    episodeDivContainer.classList.add("episode-container");
+    episodeTitleElement.classList.add("episode-title");
+    episodeImageElement.classList.add("episode-image");
+    episodeBlurbElement.classList.add("episode-blurb");
```
Comment (msimmdev) [inline script.js:162]: Praise: Great to see comments and logical grouping to make your code easier to read.
Suggestion: The way you've grouped different operations here is OK, however I think it'd be better if it was grouped per thing you are creating. So putting the episodeDivContainer , adding the class and id to it in one group and then a second group for the h3 element and so on.

File: script.js [hunk 10]
```diff
@@ -1,12 +1,121 @@
...
+
+  let defaultOption = document.createElement("option"); // step 1 createElement
+  defaultOption.textContent = "Select Episode"; // step 2 set textContent
+  dropdown.appendChild(defaultOption); // step 3 appendChild
+  dropdown.selectedIndex = 0;
+
+  for (i = 0; i < episodeList.length; i++) {
+    let option = document.createElement("option");
+    option.value = episodeList[i].name;
+    option.textContent = `S${episodeList[i].season
```
Comment (msimmdev) [inline script.js:193]: Suggestion: The logic here is good, and I like how you've chained the string operations. I do think it's a little difficult to understand as it's all inside an interpolated string. I think it'd be more readable if separated out.

e.g.
```js
let seasonNumber = episodeList[i]
  .season
  .toString()
  .padStart(2, 0);
let episodeNumber = episodeList[i]
  .number
  .toString()
  .padStart(2, 0);
option.textContent = `S${seasonNumber}E${episodeNumber}`;

File: script.js [hunk 11]
```diff
@@ -1,12 +1,121 @@
...
+      episode.summary.toLowerCase().includes(searchInput)
+    ) {
+      return episode;
+    }
+  });
+
+  document.getElementById("displaying-num").innerText = filteredEpisodes.length;
+  makePageForEpisodes(filteredEpisodes);
+
+  // BUG - will search for matches even in they are part of a longer word (eg. 'die' will also show 'soldier')
```
Comment (msimmdev) [inline script.js:237]: Question: Is that a bug?

File: style.css [hunk 12]
```diff
@@ -1,3 +1,33 @@
...
+  background-color: #ffffff;
+  border-radius: 10px;
+  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.14);
+  text-align: center;
+}
+
+.episode-title {
+  text-align: center;
+  margin-top: 1rem;
+  height: 40px;
```
Comment (msimmdev) [inline style.css:72]: Issue: You're defining some sizes in `px` and some in `rem` it's best to be consistent and use `rem` exclusively as your unit  unless there is a specific reason not to.

**Review by msimmdev — COMMENTED** [review]
Great work so far, I've noted a couple of issues that you might want to look into soon and some suggestions that you can look into if you have time.
